### PR TITLE
Porting over improvements from comparison with rnaseq pipeline

### PR DESCRIPTION
**Pull request by ewels**
I've been syncing nf-core/rnaseq with the template pipeline and found a few things in the template that needed tidying up and/or porting from the rnaseq.

Whilst I was there, I also added a new function to check the hostname against those set in nf-core/configs (see https://github.com/nf-core/configs/pull/33) and throws a warning if it looks like the correct profile was not set.

Other stuff:
* MultiQC report now added to the summary email in the template
* nf-core/config params set to null to avoid warnings if not set
* Automatically generated table of contents in markdown files (if used with Atom plugin)
* Other general tidying

### CODE REVIEW

**Comment by drpatelh** [comment]
All looks good to me.

Are you going to make any more changes @ewels ?

Just waiting for Travis CI to pass.

File: CHANGELOG.md [hunk 1]
```diff
@@ -10,7 +10,10 @@
 * Brought the logo to life
 * Change the default filenames for the pipeline trace files
 * Remote fetch of nf-core/configs profiles fails gracefully if offline
-* Remove `process.container` and just directly define `process.container` now.
+* Remove `process.container` and just directly define `process.container` now
```
[the remark this answers is not in the document]
Comment (ewels) [reply CHANGELOG.md]: Hah, serves me right for doing global find and replace..! Need to be more careful..

**Comment by ewels** [comment]
I'm not planning to make any changes for now, now. At least, not until I do the same process for other pipelines, but that'll be a different day / week / month.

Phil

**Comment by drpatelh** [comment]
The last commit to `CHANGELOG.md` is not going to affect the tests because they passed beforehand. Going to merge before Travis CI completes.

**Pull request merged**